src/main/java/org/broadinstitute/barclay/argparser/LegacyCommandLineArgumentParser.java

```diff
@@ -213,26 +209,32 @@ private LegacyCommandLineArgumentParser(final Object callerOptions, final String
             prefixDot = prefix + ".";
         }
 
-        int fieldCounter = 1;
-        for (final Field field : CommandLineParser.getAllFields(this.callerOptions.getClass())) {
+        createArgumentDefinitions(callerOptions);
+
+        this.programProperties = this.callerOptions.getClass().getAnnotation(CommandLineProgramProperties.class);
+    }
+
+    private void createArgumentDefinitions(final Object callerArguments) {
```

Comment:
I wish there a was a sane way to share this code with the other parser, since they're essentially identical now.